### ISSUE TITLE
Suggested New experiments- GPT2-small w/ Sophia on Fineweb-10B data

### ISSUE THREAD

**Issue by sanyalsunny111** [Liuhong99/Sophia#51]
Hi @Liuhong99 ,

I am a big fan of sophia used it cited it everytime. Just thought of suggesting you a new  and less resource intensive experiment.

a) Karpathy updated the nano_gpt2 training [code](https://github.com/karpathy/build-nanogpt/tree/master) with tokens w/o replacement dataloader, new data finewebedu-10B. I am curious how sophia would do given this new setting.

b) inverse layer idx used here is pretty good but recently many works have used qk normalization.

Thank you for such a good repo. Feel free to disregrad this suggestion.

**Comment by dhia680** [comment]
I trained gpt2 on finewebedu-10B with Sophia (with different settings : varying learning rates or weight decayed layers). And got this for initial results. The loss early goes up.
You can see the difference with the baseline (AdamW configured as in Nanogpt repo).
<img width="1276" alt="sophia on fineweb edu loss" src="https://github.com/user-attachments/assets/dd746c81-4ce5-4a2e-989b-2615f8b9c028">
Some suggested batch_size ramp up to solve this bad behaviour...
I haven't try this yet.
I'm open to discussions.
